crates/fluvio-connector-common/src/producer.rs

```diff
@@ -0,0 +1,41 @@
+use fluvio::FluvioConfig;
+use crate::{ConnectorConfig, Result};
+
+use crate::{ensure_topic_exists, smartmodule::smartmodule_chain_from_config};
+
+pub async fn producer_from_config(config: &ConnectorConfig) -> Result<fluvio::TopicProducer> {
+    let mut cluster_config = FluvioConfig::load()?;
+    cluster_config.client_id = Some(format!("fluvio_connector_{}", &config.name));
+
+    let fluvio = fluvio::Fluvio::connect_with_config(&cluster_config).await?;
```

Comment:
I wonder if we can somehow keep this Fluvio instance accessible and shared in the connectors so we can use the get the client metrics from that. (We can get them from the TopicProducer struct but that would not be shared if we create a consumer and a producer with different Fluvio instances

Reply:
I added a return Fluvio client along with TopicProducer and added taking metrics from it.